src/server/handler.py

```diff
@@ -43,7 +44,15 @@ async def rpc_inference(
             print("OPENED RPC_INFERENCE")
             request = await anext(requests)
             requested_uids = self._check_uids(request.uid)
+            metadata = MSGPackSerializer.loads(request.metadata) if request.metadata else {}
             requested_backends = tuple(self.module_backends[uid] for uid in requested_uids)
+            max_length = metadata.get("max_length")
+
+            if not requested_uids:
+                raise ValueError("User must specify at least one block for inference, but got none")
+            assert isinstance(max_length, int), f"rpc_inference metadata must contain int seq_length, got {max_length}"
+            if max_length not in range(0, self.inference_max_length):
```

Comment:
```suggestion
            if not 0 <= max_length <= self.inference_max_length:
```

* `max_length` should be allowed to be equal to `inference_max_length` (currently it's only allowed to be `inference_max_length - 1`)
* Double inequality is a bit better than `range()`, so there's no questions from people who are confused by Python 2 (`range()` generated the whole array there, so this line would have caused OOM)